app/src/main/java/com/example/cm/data/repositories/AuthRepository.java

```diff
@@ -14,47 +11,22 @@
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
 
-
-/**
- * inspired by https://learntodroid.com/how-to-use-firebase-authentication-in-an-android-app-using-mvvm/
- */
+import java.util.Objects;
 
 public class AuthRepository extends Repository {
-    private static final String TAG = "AuthRepository";
     private final FirebaseAuth firebaseAuth;
-    private final MutableLiveData<FirebaseUser> userLiveData;
-    private final MutableLiveData<Boolean> loggedOutLiveData;
-    private Application application = null;
-
-    public AuthRepository(Application application) {
-        this.application = application;
-        this.firebaseAuth = FirebaseAuth.getInstance();
-        this.userLiveData = new MutableLiveData<>();
-        this.loggedOutLiveData = new MutableLiveData<>();
-
-        if (firebaseAuth.getCurrentUser() != null) {
-            userLiveData.postValue(firebaseAuth.getCurrentUser());
-            loggedOutLiveData.postValue(false);
-        }
-    }
 
     public AuthRepository() {
         this.firebaseAuth = FirebaseAuth.getInstance();
-        this.userLiveData = new MutableLiveData<>();
-        this.loggedOutLiveData = new MutableLiveData<>();
-
-        if (firebaseAuth.getCurrentUser() != null) {
-            userLiveData.postValue(firebaseAuth.getCurrentUser());
-            loggedOutLiveData.postValue(false);
-        }
     }
 
     public void updatePassword(String currentPassword, String newPassword, Callback callback) {
         if (firebaseAuth.getCurrentUser() == null) {
             return;
         }
 
-        AuthCredential credential = EmailAuthProvider.getCredential(firebaseAuth.getCurrentUser().getEmail(), currentPassword);
+        AuthCredential credential = EmailAuthProvider.getCredential(Objects.requireNonNull(firebaseAuth.getCurrentUser().getEmail()), currentPassword);
```

Comment:
Add check whether currentUsers exists instead of wrapping it with requireNonNull since that could still throw an error